DOCS-20 - AI/ML Hub Page (#3793)
SumoLogic/sumologic-documentation: docs/get-started/ai-machine-learning.md

---

import useBaseUrl from '@docusaurus/useBaseUrl';

In this document, you'll learn about Sumo Logic features that leverage artificial intelligence (AI), machine learning (ML), and pattern recognition to support cloud security management, mitigate risks, reduce manual workloads for your team, and streamline incident response and resolution.

These alerting, security, and remediation features go beyond visual analytics. They are tools that accelerate issue resolution, reduce mean time to respond (MTTR), and empower you to proactively monitor and safeguard your technology stack against evolving threats.

<details>
<summary>What's the difference?</summary>
* Artificial intelligence
* Machine learning
* Pattern recognition

</details>

<!-- Remove OR ask brand to create a cleaner, branded version
<img src={useBaseUrl('img/get-started/ai-diagram.png')} alt="ai-diagram.png" />
-->

## Observability

Sumo Logic provides observability into your technology stack by analyzing the millions of log files created by your technology stack, detecting anomalies and outlier data, and reporting security issues in a timely fashion with fewer false positives.

### Log Search

> * [Copilot](/docs/search/copilot). Our in-product AI-based assistant that transforms log analysis by enabling users to quickly obtain actionable insights. Through plain English queries and automatic log query generation, Copilot simplifies the investigation process, allowing even users without extensive log analysis expertise to pinpoint anomalies and potential threats efficiently.
> 
> With Copilot, users can effortlessly investigate complex issues without writing intricate log queries manually. Its intuitive interface guides users through each step of the investigation, refining queries based on AI prompts and feedback. This streamlined approach accelerates the identification of security threats, empowering users to make informed decisions rapidly and proactively detect potential risks.

Author: ```suggestion
* [Copilot](/docs/search/copilot). Our in-product AI-based assistant that transforms log analysis by enabling users to quickly obtain actionable insights. Through plain English queries and automatic log query generation, Copilot simplifies the investigation process, allowing even users without extensive log analysis expertise to pinpoint anomalies and potential threats efficiently. With Copilot, users can effortlessly investigate complex issues without writing intricate log queries manually. Its intuitive interface guides users through each step of the investigation, refining queries based on AI prompts and feedback. This streamlined approach accelerates the identification of security threats, empowering users to make informed decisions rapidly and proactively detect potential risks.
```

### Log Search

* [Copilot](/docs/search/copilot). An in-product AI-based assistant that allows you to obtain insights quickly from logs. You can interact with Copilot using plain English queries and utilize search suggestions for result refinement.
* [LogReduce&reg;](/docs/search/logreduce). Utilizes AI-driven algorithms to cluster log messages based on string similarity, and boils down thousands of log lines into easy-to-understand patterns. Separate the signal from the noise and detect anomalous behavior with Outlier Detection. LogReduce employs fuzzy logic to group similar messages into signatures, enabling quick assessment of activity patterns. You can refine results based on your preferences, [teaching LogReduce](/docs/search/logreduce/influence-the-logreduce-outcome) for more specific outcomes.
* [LogCompare](/docs/search/logcompare). Identify and predict anomalies in real time with outlier detection, and uncover root-causes using LogCompare pattern analysis.

### Alerts

#### Anomaly Detection

> [Anomaly Detection](/docs/alerts/monitors/create-monitor/#select-monitor-type-and-detection-method) applies ML techniques to detect anomalies and identify suspicious activity. When you create a monitor using this method, it establishes a baseline for normal signal behavior, leveraging historical data to minimize false positives. Key features include:
>   - Detection of seasonality in log signals (hourly, daily, weekly).
>   - Auto-tuned anomaly detection for minimal user input.
>   - Extensible detector framework allowing you to specify additional context for anomaly detection.
>   - Associate anomaly response with a monitor by automating a playbook.
> 
> AI-driven alerting overcomes monitoring limitations through:
> 
> * **Model-driven anomaly detection**. Utilizing historical data, ML models establish accurate baselines, eliminating guesswork and noise in alerts.
> * **AutoML**. The system self-tunes, including seasonality detection, minimizing user intervention for a simpler experience.
> * **User context**. Users set alert sensitivity and incident thresholds, adding context to anomaly detection to mitigate noise.
> * **One-click playbook assignment**. Monitors seamlessly link to Automation Service playbooks, expediting response without manual intervention.
> * **Auto-diagnosis and recovery**. Sumo Logic Automation Service automates diagnosis and resolution, closing the loop from alert to recovery.

Author: ```suggestion
[Anomaly Detection](/docs/alerts/monitors/create-monitor/#select-monitor-type-and-detection-method), powered by machine learning, efficiently flags suspicious activities by establishing baseline behavior and minimizing false positives. It also automatically fine-tunes anomaly detection with minimal user input, and you can associate it with a playbook to link anomaly responses with monitors, streamlining incident response.

AI-driven alerting enhances monitoring by:
- **Model-driven Anomaly Detection**. Utilizes historical data to create accurate baselines, reducing noise.
- **AutoML**. Self-tunes, including detecting seasonality, requiring less user intervention.
- **One-click Playbook Assignment**. Links monitors to Automation Service playbooks for faster response.
- **Auto-diagnosis and recovery**. Automation Service automates diagnosis and resolution, completing the alert-to-recovery process.
```

#### Automated playbooks

> With our [Automated playbooks](/docs/alerts/monitors/use-playbooks-with-monitors), you can set up a predefined set of actions and conditional statements that respond to an events like security incidents by running an automated workflow. Configuration is easy - browse our 500+ existing playbooks in the Automation Service App Central, then choose and/or customize it.

Author: ```suggestion
With [Automated playbooks](/docs/alerts/monitors/use-playbooks-with-monitors), you can set up a predefined set of actions and conditional statements that respond to an events like security incidents proactively by running an automated workflow without manual intervention. Configuration is easy - browse our 500+ existing playbooks in the Automation Service App Central, then choose and/or customize it. You can access playbooks when creating a monitor, viewing an alert, or directly from the Automation Service.
```

<details>
<summary>What do these terms mean?</summary>

AI encompasses machines that mimic human-like intelligence, leveraging algorithms to compute tasks efficiently. It includes machine learning and deep learning.

ML, a subset of AI, involves training machines to learn from data without explicit programming, improving performance over time. Within ML, there are various types: supervised learning, unsupervised learning, semi-supervised learning, and reinforcement learning, each suited for different problem settings such as classification, regression, and clustering.

Deep learning, another subset of AI, employs artificial neural networks with multiple layers to process data, excelling in tasks like image recognition and natural language understanding. Generative AI, closely related to deep learning, produces new data resembling training data, often utilizing large language models like GPT.

Pattern learning is fundamental to machine learning and deep learning, where algorithms discern patterns in data to make predictions or classifications, enabling advancements in various fields through data-driven decision-making.

</details>

> Our alerting, security, and remediation features go beyond visual analytics. They are tools that accelerate issue resolution, reduce mean time to respond (MTTR), and empower you to proactively monitor and safeguard your technology stack against evolving threats.

Author: ```suggestion
Our alerting, security, and remediation features extend beyond visual analytics, providing essential tools to accelerate issue resolution, reduce mean time to respond (MTTR), and empower proactive monitoring and safeguarding of your technology stack against evolving threats.
```

<!-- Remove or ask brand to create a cleaner, branded version
<img src={useBaseUrl('img/get-started/ai-diagram.png')} alt="ai-diagram.png" />
-->

## Observability

> Sumo Logic provides observability into your technology stack by analyzing the millions of log files created by your technology stack, detecting anomalies and outlier data, and reporting security issues in a timely fashion with fewer false positives.

Reviewer: ```suggestion
Sumo Logic provides observability into your technology stack by analyzing the millions of log files created in your environment, detecting anomalies and outlier data, and reporting security issues in a timely fashion with fewer false positives.
```

* [Microsoft Sentinel](/docs/platform-services/automation-service/app-central/integrations/microsoft-sentinel)
* [Darktrace](/docs/platform-services/automation-service/app-central/integrations/darktrace)
* [Criminal IP](/docs/platform-services/automation-service/app-central/integrations/criminal-ip)
* [Arcanna](/docs/platform-services/automation-service/app-central/integrations/arcanna)

<!-- Removed - not in Kanso
### Root Cause Explorer

Accelerates troubleshooting by surfacing events of interest correlated with application incidents. [Learn more](/docs/observability/root-cause-explorer)
-->

## Security

### Cloud SIEM

Sumo Logic's Cloud SIEM leverages AI-driven rules for security management, including:

* [Insight Trainer](/docs/cse/rules/insight-trainer). Utilizes ML and AI to deliver outcome-based recommendations, reducing false positives without compromising detection value.
* [First-seen rule](/docs/cse/rules/write-first-seen-rule). Identifies novel threats based on first occurrences.
* [Outlier rule](/docs/cse/rules/write-outlier-rule/). Detects abnormal behavior indicating potential security breaches.

### Global Intelligence

> Our Global Intelligence Service apps provide security teams with valuable real-time security intelligence to scale detection, prioritization, investigation, and workflow to prevent potentially harmful service configurations that could lead to a costly data breach.
> 
> * [Global Intelligence Service](/docs/integrations/global-intelligence)

Author: ```suggestion
Our Global Intelligence Service apps provide security teams with valuable real-time security intelligence to scale detection, prioritization, investigation, and workflow to prevent potentially harmful service configurations that could lead to a costly data breach. [Learn more](/docs/integrations/global-intelligence).
```

---

import useBaseUrl from '@docusaurus/useBaseUrl';

In this document, you'll learn about Sumo Logic features that leverage artificial intelligence (AI), machine learning (ML), and pattern recognition to support cloud security management, mitigate risks, reduce manual workloads for your team, and streamline incident response and resolution.

These alerting, security, and remediation features go beyond visual analytics. They are tools that accelerate issue resolution, reduce mean time to respond (MTTR), and empower you to proactively monitor and safeguard your technology stack against evolving threats.

<details>
<summary>What's the difference?</summary>
* Artificial intelligence
* Machine learning
* Pattern recognition

</details>

<!-- Remove OR ask brand to create a cleaner, branded version
<img src={useBaseUrl('img/get-started/ai-diagram.png')} alt="ai-diagram.png" />
-->

## Observability

Sumo Logic provides observability into your technology stack by analyzing the millions of log files created by your technology stack, detecting anomalies and outlier data, and reporting security issues in a timely fashion with fewer false positives.

### Log Search

> * [Copilot](/docs/search/copilot). An in-product AI-based assistant that allows you to obtain insights quickly from logs. You can interact with Copilot using plain English queries and utilize search suggestions for result refinement.

Author: ```suggestion
* [Copilot](/docs/search/copilot). Our in-product AI-based assistant that transforms log analysis by enabling users to quickly obtain actionable insights. Through plain English queries and automatic log query generation, Copilot simplifies the investigation process, allowing even users without extensive log analysis expertise to pinpoint anomalies and potential threats efficiently.

With Copilot, users can effortlessly investigate complex issues without writing intricate log queries manually. Its intuitive interface guides users through each step of the investigation, refining queries based on AI prompts and feedback. This streamlined approach accelerates the identification of security threats, empowering users to make informed decisions rapidly and proactively detect potential risks.
```

> * [Copilot](/docs/search/copilot). Our in-product AI-based assistant that transforms log analysis by enabling users to quickly obtain actionable insights. Through plain English queries and automatic log query generation, Copilot simplifies the investigation process, allowing even users without extensive log analysis expertise to pinpoint anomalies and potential threats efficiently. With Copilot, users can effortlessly investigate complex issues without writing intricate log queries manually. Its intuitive interface guides users through each step of the investigation, refining queries based on AI prompts and feedback. This streamlined approach accelerates the identification of security threats, empowering users to make informed decisions rapidly and proactively detect potential risks.
> * [LogReduce&reg;](/docs/search/logreduce). Utilizes AI-driven algorithms to cluster log messages based on string similarity, and boils down thousands of log lines into easy-to-understand patterns. Separate the signal from the noise and detect anomalous behavior with Outlier Detection. LogReduce employs fuzzy logic to group similar messages into signatures, enabling quick assessment of activity patterns. You can refine results based on your preferences, [teaching LogReduce](/docs/search/logreduce/influence-the-logreduce-outcome) for more specific outcomes.
> * [LogCompare](/docs/search/logcompare). Identify and predict anomalies in real time with outlier detection, and uncover root-causes using LogCompare pattern analysis.

Author: ```suggestion

#### Copilot

Copilot, our in-product AI-based assistant, transforms log analysis by enabling users to quickly obtain actionable insights. Through plain English queries and automatic log query generation, Copilot simplifies the investigation process, allowing even users without extensive log analysis expertise to pinpoint anomalies and potential threats efficiently.

With Copilot, users can effortlessly investigate complex issues without writing intricate log queries manually. Its intuitive interface guides users through each step of the investigation, refining queries based on AI prompts and feedback. This streamlined approach accelerates the identification of security threats, empowering users to make informed decisions rapidly and proactively detect potential risks. [Learn more](/docs/search/copilot).

#### LogReduce

LogReduce&reg; utilizes AI-driven algorithms to cluster log messages based on string similarity, and boils down thousands of log lines into easy-to-understand patterns. Separate the signal from the noise and detect anomalous behavior with Outlier Detection. LogReduce employs fuzzy logic to group similar messages into signatures, enabling quick assessment of activity patterns. You can refine results based on your preferences, teaching LogReduce for more specific outcomes. [Learn more](/docs/search/logreduce).

#### LogCompare

LogCompare offers a streamlined solution for comparing log data across different time periods to detect changes or anomalies efficiently. By analyzing log patterns and generating delta analysis, LogCompare aids in troubleshooting and root cause discovery, empowering you to identify significant changes in your log messages.

Through its intuitive interface, LogCompare simplifies the comparison process, allowing you to create baseline queries and compare them to current queries effortlessly. By highlighting differences and providing variance reports, LogCompare enables you to pinpoint anomalies and track the evolution of log patterns over time, facilitating proactive threat detection and rapid decision-making. [Learn more](/docs/search/logcompare).
```
